plugins/modules/clickhouse_quota.py

```diff
@@ -0,0 +1,644 @@
+    type: str
+  keyed_by:
+    description:
+      - Keys the quota by the specified key (default is to not key)
+    type: str
+    choices:
+      - user_name
+      - ip_address
+      - client_key
+      - client_key,user_name
+      - client_key,ip_address
+  limits:
+    description:
+      - The limits that this quota should enforce.
+    type: list
+    elements: dict
+    suboptions:
+      randomized_start:
+        description:
+          - Whether this interval's start should be randomized.
+          - Intervals always start at the same time if not randomized.
+        type: bool
+        default: false
+      interval:
+        description:
+          - The interval to apply the following quotas on
+          - This is in the format <number> <unit>
+          - Where unit is one of second, minute, hour, day, week, month, quarter or year
+        type: str
+        required: true
+      max:
+        description:
+          - Maximum values to apply to this interval in this quota
+          - At least one key must be specified
+          - Mutually exclusive with I(no_limits) and I(tracking_only)
+        type: dict
+        suboptions:
+          queries:
+            description:
+              - Maximum number of queries to enforce in this interval
+            type: int
+          query_selects:
+            description:
+              - Maximum number of query selects to enforce in this interval
+            type: int
+          query_inserts:
+            description:
+              - Maximum number of query inserts to enforce in this interval
+            type: int
+          errors:
+            description:
+              - Maximum number of errors to enforce in this interval
+            type: int
+          result_rows:
+            description:
+              - Maximum number of result rows to enforce in this interval
+            type: int
+          result_bytes:
+            description:
+              - Maximum number of result bytes to enforce in this interval
+            type: int
+          read_rows:
+            description:
+              - Maximum number of rows read to enforce in this interval
+            type: int
+          read_bytes:
+            description:
+              - Maximum number of bytes read to enforce in this interval
+            type: int
+          written_bytes:
+            description:
+              - Maximum number of bytes written to enforce in this interval
+            type: int
+          execution_time:
+            description:
+              - Maximum number of execution time to enforce in this interval
+            type: float
+          failed_sequential_authentications:
+            description:
+              - Maximum number of failed sequential authentications to enforce in this interval
+            type: int
+      no_limits:
+        description:
+          - Don't apply any limits
+          - Mutually exclusive with I(max) and I(tracking_only)
```

Comment:
```suggestion
          - Don't apply any limits.
          - Mutually exclusive with O(max) and O(tracking_only).
```
Also in the style guide, it's recommended to use the long form for things like "Don't" -> "Do not". FYI